Swifth/Layers/Bluetooth/PeripheralManager/PeripheralManager.swift

```
@@ -63,20 +63,47 @@ class PeripheralManager : NSObject {
 
 extension PeripheralManager {
     
-    func updateTemperatureService(withData data: Data) {
+    private func updateCharacteristic(_ characteristic: CBMutableCharacteristic, withData data: Data) {
+        characteristic.value = data
+        if self.cbPeripheralManager?.updateValue(data, for: characteristic, onSubscribedCentrals: nil) == false {
+            // FIXME: report an error
+        }
+    }
+    
+    private func service() throws -> TemperatureService {
         guard let sensor = self.sensor else {
             assertionFailure("invalid state of \(self)")
-            return
+            throw NSError(domain: "", code: 1, userInfo: nil)
```

would be better to throw an actual error here.